ecscollector/collector.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ecscollector implements a Prometheus collector for Amazon ECS
+// metrics available at the ECS metadata server.
+package ecscollector
+
+import (
+	"context"
+	"log"
+
+	"github.com/prometheus-community/ecs_exporter/ecsmetadata"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	numProcsDesc = prometheus.NewDesc(
+		"ecs_num_procs",
```

Comment:
Simplify to `ecs_processes`.

Reply:
Removed this metric for now, it returns 0 for each request and seems like it's broken. We will take a look and can introduce it back.